position/views.py

```diff
@@ -0,0 +1,79 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpResponseRedirect
+from django.urls import reverse_lazy
+from django.utils import timezone
+from django.views.generic import ListView, CreateView, UpdateView, DeleteView
+
+from dudes9 import settings
+from relation.forms import UpdatePositionForm, AddRelationForm
+from relation.models import Relation
+
+
+class PositionUpdateView(UpdateView):
+    model = Relation
+    template_name = 'position_update.html'
+    form_class = UpdatePositionForm
+    success_url = reverse_lazy('relations')
+
+    def post(self, request, *args, **kwargs):
+        time = str(timezone.now())
+
+        self.object = self.get_object()
+
+        old_relation = Relation.objects.get(pk=self.object.pk)
+        old_relation.date_left = time
+        old_relation.save()
+
+        new_relation = Relation(member=self.object.member, position=request.POST['position'],
+                                project=self.object.project, date_joined=time)
+        new_relation.save()
+        return HttpResponseRedirect(self.get_success_url())
+
+
+class AddRelationView(CreateView):
+    model = Relation
+    form_class = AddRelationForm
+    template_name = 'add_relation_form.html'
+    success_url = reverse_lazy('relations')
+
+    def post(self, request, *args, **kwargs):
+        time = str(timezone.now())
+        self.object = None
+
+        try:
+            relation = Relation.objects.get(member_id=request.POST['member'], project=request.POST['project'],
```

Comment:
just a note:
if you will do Relation.objects.filter(....).first() it will return your object(if it exists) or None. Simple one line without try/except